docs/DeveloperGuide.md

```diff
@@ -1,34 +1,956 @@
+    - Cons: It is unoptimized in terms of complexity for search operations, which requires more work for scaling of the
+    application.
+- **Alternative 2:** Store as a Hash Table with the key as the tag and value as `Task`
+    - Pros: It has a better time complexity for search operations since this data structure is more optimized (O(1) can
+    be achieved).
+    - Cons: It is hard to retrieve the tag for a specific `Task` due to the structure of the key-value pair.
+
+#### 4.2.3. List Feature
+
+##### 4.2.3.1. Implementation
+
+The following sequence diagram shows how the list operation works:
+
+![Sequence Diagram for List command](/diagrams/class/jpeg/SequenceDiagram_List.jpg)
+
+### 4.2.3. Mark as done feature  
+
+#### 4.2.2.1. Implementation  
+
+The following sequence diagram hows how the mark as done operation works:  
+
+#### 4.2.2.2. Design Considerations  
+
+### 4.3. Notebook Mode
+
+#### 4.3.1. Notebook Management Feature
+
+As shown in Figure 1, the `NotebookShelf` class comprises instances of `Notebook` class. `Notebook` comprises `Section`
+and `Section` comprises `Page`. The navigability is not bidirectional. Multiple operations such as addition and deletion
+can be done without affecting other instances at all, while updating the `Notebook` it is in.
+
+This section explains the implementation and design considerations for managing `Notebook`s.
+
+##### 4.3.1.1 Implementation
+There are two main functions in notebook management: add and remove.
+<br>
+
+Given below is an example usage scenario and how the add notebook function behaves.
+
+Step 1. The user launches the application for the first time. `CliUserInterface#executeCommand` is called when the user adds new `Notebook` into the `NotebookShelf`.
+
+Step 2. The user types `add /nCS2113T`. The `add` command is passed through `InputParser#getCommandFromInput`.
+
+Step 3. `InputParser#parseNotebookTitle` is then called, and it returns the `titleToAdd`, which is `CS2113T`.
+
+Step 4. A constructor for `AddCommandNotebookMode` is created.
+
+Step 5. `AddCommandNotebookMode#execute()` runs, which then calls `NotebookShelf#addNotebook`.
+
+Step 6. A new `Notebook`, entitled `CS2113T` is initialised.
+
+The UML sequence diagram below shows how the add task command works.
+![Sequence Diagram for Add Task Command](/diagrams/class/jpeg/add_notebook.jpg)
+
+<hr>
+
+The figure below shows how the "remove task" command works:
+<img src="https://user-images.githubusercontent.com/60319628/96821973-9176e600-145b-11eb-95b7-5bf885ea1867.png">
+
+After calling `InputParser#getCommandFromInput` from `CliUserInterface`:
+1. `InputParser` parses the input to return the `notebookTitleToRemove`, `sectionTitleToRemove` and `pageNumberToRemove`.
+Some of these members may be empty.
+2. `InputParser` constructs and returns the `RemoveCommandNotebookMode` class with constructor as shown below:
+```java
+public RemoveCommandNotebookMode(String notebookTitle, String sectionTitle,
```

Comment:
The code snippet is nice to have, but I don't feel like it adds that much to the description, and if the code changes then this needs to be changed too.

Reply:
Got it, thanks.